src/Orleans.Core/Streams/Internal/IInternalAsyncObservable.cs

```diff
@@ -18,7 +23,7 @@ internal interface IInternalAsyncObservable<T> : IAsyncObservable<T>
     }
 
         
-    internal interface IInternalAsyncBatchObserver<in T> : IAsyncBatchObserver<T>
+    internal interface IInternalAsyncBatchObserver<T> : IAsyncObserver<T>
```

Comment:
should be this `IAsyncBatchObserver` instead of `IAsyncObserver` ?

Reply:
I opted to inherit from only IAsyncObserver since the feature request was only for batched reads, not batched writes, but that was due to the below issues, which I've not yet found a 'good' solution.

IAsyncBatchObserver  and IAsyncObserver share functions, so inheriting from both interfaces cause conflicts.  We can work around them, but kinda sucks.
Original batch observer inherited from observer to avoid this, but that requires implementors to implement both batch and not batch calls and makes it unclear which will be called.
We could take the common functions and include them in a shared interfaces so their' not common, but that changes existing IAsyncObserver interface, which I was trying to avoid.